rewrite-maven/src/test/java/org/openrewrite/maven/search/DoesNotIncludeDependencyTest.java

```diff
@@ -15,532 +15,472 @@
-                        <artifactId>spring-beans</artifactId>
-                        <version>6.0.0</version>
-                        <scope>%s</scope>
-                      </dependency>
-                    </dependencies>
-                  </project>
-                  """, scope),
-                String.format("""
-                  <!--~~>--><project>
-                    <modelVersion>4.0.0</modelVersion>
-                    <groupId>org.sample</groupId>
-                    <artifactId>sample</artifactId>
-                    <version>1.0.0</version>
-                    <dependencies>
-                      <dependency>
-                        <groupId>org.springframework</groupId>
-                        <artifactId>spring-beans</artifactId>
-                        <version>6.0.0</version>
-                        <scope>%s</scope>
-                      </dependency>
-                    </dependencies>
-                  </project>
-                  """, scope)
-              )
-            );
+            @Test
+            void multimoduleMarksOnlyCorrectModule() {
+                rewriteRun(
+                  spec -> spec.recipe(defaultRecipeWithOnlyDirectAndScope(null, null)),
+                  pomXml(
+                    String.format(transitiveDependencyTemplate, ""),
+                    spec -> spec.path("a/pom.xml")
+                  ),
+                  pomXml(
+                    missingDependencyTemplate,
+                    marker + missingDependencyTemplate,
+                    spec -> spec.path("b/pom.xml")
+                  ),
+                  pomXml(
+                    missingDependencyTemplate,
+                    marker + missingDependencyTemplate
+                  )
+                );
+            }
         }
 
-        @ParameterizedTest
-        @ValueSource(strings = {"compile", "runtime", "test"})
-        void dependencyPresentSpecifiedTestScopeNotMarked(String scope) {
-            rewriteRun(
-              spec -> spec.recipe(new DoesNotIncludeDependency("org.springframework", "spring-beans", true, "test")),
-              pomXml(
-                String.format("""
-                  <project>
-                    <modelVersion>4.0.0</modelVersion>
-                    <groupId>org.sample</groupId>
-                    <artifactId>sample</artifactId>
-                    <version>1.0.0</version>
-                    <dependencies>
-                      <dependency>
-                        <groupId>org.springframework</groupId>
-                        <artifactId>spring-beans</artifactId>
-                        <version>6.0.0</version>
-                        <scope>%s</scope>
-                      </dependency>
-                    </dependencies>
-                  </project>
-                  """, scope)
-              )
-            );
+        @Nested
+        class OnlyDirectTrue {
+            @DocumentExample
```

Comment:
The way `@DocumentExample` works is that we look at the `.java` source file, and extract the values without running the tests. In this case I think it wouldn't work, so then it's maybe best to add an example for the docs at the top that's just a single straightforward two text blocks test.